src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java

```diff
@@ -124,7 +125,11 @@ public boolean verifyCertificate(X509Certificate certificate, Attributes attribu
             Attribute attribute = attributes.get(ldapAttribute);
             final int size = attribute.size();
             for (int i = 0; i < size; i++) {
-                if (certificate.getSubjectDN().getName().equals(attribute.get(i))) {
+
+                X500Principal certSubjectX500Principal = certificate.getSubjectX500Principal();
+                X500Principal attSubjectX500Principal = new X500Principal((String)attribute.get(i));
```

Comment:
One further point - when looking at others it is not just about which ones are retrieved but you also need to double check how it is used.

As an example the code you changes previously passed the value straight to an equals method, in that case passing in null was safe as equals would (assuming it is correctly written) detect that and return false.